kernel/src/snapshot.rs

```diff
@@ -472,6 +481,47 @@ mod tests {
         assert!(cp.is_none())
     }
 
+    fn valid_last_checkpoint() -> Vec<u8> {
+        r#"{"size":8,"size_in_bytes":21857,"version":1}"#.as_bytes().to_vec()
+    }
+
+    #[test]
+    fn test_read_table_with_invalid_last_checkpoint() {
+        // in memory file system
+        let store = Arc::new(InMemory::new());
+
+        // put _last_checkpoint file
+        let data = valid_last_checkpoint();
+        let invalid_data = "invalid".as_bytes().to_vec();
+        let path = Path::from("valid/_last_checkpoint");
+        let invalid_path = Path::from("invalid/_last_checkpoint");
+
+        tokio::runtime::Runtime::new()
```

Comment:
can all of this be replace with a `#[tokio::test]` instead?

Reply:
yea considered doing that but figured this called out async a little better so kept it as-is. I would prefer to have these be sync tests but I think we are forced to async with object_store..

Reply:
Does calling the async out here provide some value to someone trying to understand the test? Assuming they know rust async, doing the runtime plumbing just seems like boiler plate right?

Reply:
probably not. I would like to think about how to isolate the async code (really just needed for setup) so this is more of a TODO